File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] result=new int[nums.length];
        for(int e=0,f=0,i=0,j=1;e<nums.length;i+=2,j+=2,e++,f++){
            while(e<nums.length && nums[e]<0) e++;
            if(e==nums.length) break;
            while(f<nums.length && nums[f]>=0) f++;
            if(f==nums.length) break;
            result[i]=nums[e];
            result[j]=nums[f];
        }
        return result;
    }
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n=nums.size();
        vector<int> result(n);
        for(int e=0,f=0,g=1;e<n;e++){
            if(nums[e]<0){
                result[g]=nums[e];
                g+=2;
            }else{
                result[f]=nums[e];
                f+=2;
            }
        }
        return result;
    }
};